sfepy/discrete/fem/mappings.py

```diff
@@ -11,9 +11,52 @@
 from sfepy.linalg.utils import invs_fast, dets_fast
 
 
-def eval_mapping_data_in_qp(coors, conn, dim, bf_g, weights,
+def tranform_coors_to_lower_dim(coors, to_dim):
+    """
+    Transform element coordinates into XY plane.
+
+    See:
+      https://math.stackexchange.com/questions/1167717/transform-a-plane-to-the-xy-plane
+      https://en.wikipedia.org/wiki/Rotation_matrix#Rotation_matrix_from_axis_and_angle
+    """
+    if coors.shape[-1] == 3 and to_dim == 2:
+        bv = nm.cross(coors[:, 1, :] - coors[:, 0, :],
+                      coors[:, -1, :] - coors[:, 0, :])
+
+        b = nm.linalg.norm(bv, axis=1)
+        ab = nm.sqrt(bv[:, 0]**2 + bv[:, 1]**2)
+        cphi = bv[:, 2] / b
+        sphi = ab / b
+
+        mtxR = nm.eye(3, 3) * cphi[:, None, None]
+        idxs = nm.where(nm.abs(ab) > 1e-16)[0]
```

Comment:
`sfepy.linalg.geometry.make_axis_rotation_matrix()` ?

Reply:
I need to check this part of the code once again and I should create some tests.